gpdb/gpinitsystem.go

```diff
@@ -48,6 +47,27 @@ func (i *Installation) generatePortRange() {
 	}
 }
 
+//
+func (i *Installation) checkPGConfigFile() bool {
+
+	i.PostgresConfFileLocation = fmt.Sprintf("%s%s",Config.INSTALL.PGCONFDIRECTORY, "postgresql.conf")
+
+	Infof("Checking for custom postgresql.conf at: %s", i.PostgresConfFileLocation)
+
+	pgConfExists, err := doesFileOrDirExists(i.PostgresConfFileLocation)
+	if err != nil {
+		Errorf("Cannot get the information of directory %s, err: %v", i.PostgresConfFileLocation, err)
+	}
+
+	if !pgConfExists {
```

Comment:
```suggestion
	if Config.CORE.DATALABS && !pgConfExists {
```

Something like this should do it, if you agree @faisaltheparttimecoder ?

Reply:
This make sense, I forgot that you had an another check on the tool....

Reply:
Actually, this should not go in here, but in `executeGpInitSystem()`.